src/services/email/email.ts

```diff
@@ -0,0 +1,73 @@
+import nodemailer from "nodemailer"
+import Config from "../../config"
+import Logger from "../../logger"
+import Mail from "nodemailer/lib/mailer";
+import Result from "../../result";
+
+export class Emailer {
+    private static instance: Emailer
+    private _mail: Mail
+
+    public static get Instance(): Emailer {
+        if(!Emailer.instance) {
+            Emailer.instance = new Emailer();
+        }
+
+        return Emailer.instance
+    }
+
+    constructor() {
+        const auth: {[key:string]: any} = {};
+
+        // authentication could be LOGIN or OAUTH2
+        if(Config.smtp_username !== "") auth.user = Config.smtp_username
```

Comment:
Do we need to include any error handling in this section in case the Config values have not been set (for login or oauth2)?

Reply:
the error logging for when the connection is being made should handle any issues with bad configuration.